File: src/main/java/com/coremal/wconnector/app/service/enums/MotivoBloqueioEnum.java
package com.coremal.wconnector.app.service.enums;

public enum MotivoBloqueioEnum {
    BLQ_LIM_CREDITO("2-LIM.CREDITO"),
    BLQ_DUP_ATRASO_2D("3-DUP.ATRASO.2D"),
    BLQ_VENDA_VISTA("6-VENDA.VISTA"),
    BLQ_MB_MEDIO_MENOR_8_PERC("1-MB.MEDIO<8%"),
    BLQ_MB_REPOS_MENOR_8_PERC("1-MB.REPOS<8%"),
    BLQ_MB_MEDIO_MENOR_16_PERC("1-MB.MEDIO<16%"),
    BLQ_MB_REPOS_MENOR_16_PERC("1-MB.REPOS<16%"),
    BLQ_MB_MEDIO_MENOR_20_PERC("1-MB.MEDIO<20%"),
    BLQ_MB_REPOS_MENOR_20_PERC("1-MB.REPOS<20%");

    private final String valor;

    MotivoBloqueioEnum(String valor) {
        this.valor = valor;
    }

    public String getValor() {
        return valor;
    }

    public static MotivoBloqueioEnum porValor(String valor) {
        for (MotivoBloqueioEnum motivo : MotivoBloqueioEnum.values()) {
            if (motivo.getValor().equalsIgnoreCase(valor)) {
                return motivo;
            }
        }

        throw new IllegalArgumentException("Motivo de bloqueio inválido: " + valor);
    }
}

File: src/main/webapp/app/models/Pedido/motivo-bloqueio-enum.ts
export enum MotivoBloqueioEnum {
  BLQ_LIM_CREDITO = 'BLQ_LIM_CREDITO',
  BLQ_DUP_ATRASO_2D = 'BLQ_DUP_ATRASO_2D',
  BLQ_VENDA_VISTA = 'BLQ_VENDA_VISTA',
  BLQ_MB_MEDIO_MENOR_8_PERC = 'BLQ_MB_MEDIO_MENOR_8_PERC',
  BLQ_MB_REPOS_MENOR_8_PERC = 'BLQ_MB_REPOS_MENOR_8_PERC',
  BLQ_MB_MEDIO_MENOR_16_PERC = 'BLQ_MB_MEDIO_MENOR_16_PERC',
  BLQ_MB_REPOS_MENOR_16_PERC = 'BLQ_MB_REPOS_MENOR_16_PERC',
  BLQ_MB_MEDIO_MENOR_20_PERC = 'BLQ_MB_MEDIO_MENOR_20_PERC',
  BLQ_MB_REPOS_MENOR_20_PERC = 'BLQ_MB_REPOS_MENOR_20_PERC',
}

export const MotivoBloqueioEnumMapper: Record<string, string> = {
  [MotivoBloqueioEnum.BLQ_LIM_CREDITO]: '2-LIM.CREDITO',
  [MotivoBloqueioEnum.BLQ_DUP_ATRASO_2D]: '3-DUP.ATRASO.2D',
  [MotivoBloqueioEnum.BLQ_VENDA_VISTA]: '6-VENDA.VISTA',
  [MotivoBloqueioEnum.BLQ_MB_MEDIO_MENOR_8_PERC]: '1-MB.MEDIO<8%',
  [MotivoBloqueioEnum.BLQ_MB_REPOS_MENOR_8_PERC]: '1-MB.REPOS<8%',
  [MotivoBloqueioEnum.BLQ_MB_MEDIO_MENOR_16_PERC]: '1-MB.MEDIO<16%',
  [MotivoBloqueioEnum.BLQ_MB_REPOS_MENOR_16_PERC]: '1-MB.REPOS<16%',
  [MotivoBloqueioEnum.BLQ_MB_MEDIO_MENOR_20_PERC]: '1-MB.MEDIO<20%',
  [MotivoBloqueioEnum.BLQ_MB_REPOS_MENOR_20_PERC]: '1-MB.REPOS<20',
};

export const MotivoBloqueioEnumDescricaoMapper: Record<string, MotivoBloqueioEnum> = {
  ['2-LIM.CREDITO']: MotivoBloqueioEnum.BLQ_LIM_CREDITO,
  ['3-DUP.ATRASO.2D']: MotivoBloqueioEnum.BLQ_DUP_ATRASO_2D,
  ['6-VENDA.VISTA']: MotivoBloqueioEnum.BLQ_VENDA_VISTA,
  ['1-MB.MEDIO<8%']: MotivoBloqueioEnum.BLQ_MB_MEDIO_MENOR_8_PERC,
  ['1-MB.REPOS<8%']: MotivoBloqueioEnum.BLQ_MB_REPOS_MENOR_8_PERC,
  ['1-MB.MEDIO<16%']: MotivoBloqueioEnum.BLQ_MB_MEDIO_MENOR_16_PERC,
  ['1-MB.REPOS<16%']: MotivoBloqueioEnum.BLQ_MB_REPOS_MENOR_16_PERC,
  ['1-MB.MEDIO<20%']: MotivoBloqueioEnum.BLQ_MB_MEDIO_MENOR_20_PERC,
  ['1-MB.REPOS<20']: MotivoBloqueioEnum.BLQ_MB_REPOS_MENOR_20_PERC,
};
